pilot/pkg/proxy/envoy/v2/cds.go

```diff
@@ -78,7 +78,29 @@ func (s *DiscoveryServer) generateRawClusters(node *model.Proxy, push *model.Pus
 		return nil, err
 	}
 
+	if sdsTokenPath, found := node.Metadata[model.NodeMetadataSdsTokenPath]; found && len(sdsTokenPath) > 0 {
+		// If SDS_TOKEN_PATh is in the node metadata, make a copy of rawClusters so that
+		// the path of SDS token will be applied to the copied clusters.
+		clusters := make([]*xdsapi.Cluster, 0)
+		for _, c := range rawClusters {
+			bytes, err := c.Marshal()
+			if err != nil {
+				adsLog.Warnf("Error when marshal cluster: %v, error: %v", c, err)
+				continue
+			}
+			cp := &xdsapi.Cluster{}
+			err = cp.Unmarshal(bytes)
+			if err != nil {
+				adsLog.Warnf("Error when unmarshal cluster, error: %v", err)
+				continue
+			}
+			clusters = append(clusters, cp)
+		}
+		rawClusters = clusters
```

Comment:
So make a copy and throw it away ? I guess the cache is safe for thread access - but I can't see how memory allocs will perform.

/hold - this must be perf-tested before merging to a stable branch. 
@duderino 
@mandarjog